src/Type/Generic/TemplateTypeHelper.php

```diff
@@ -85,8 +86,35 @@ public static function resolveToBounds(Type $type): Type
 	 */
 	public static function toArgument(Type $type): Type
 	{
+		$ownedTemplates = [];
+
+		if ($type instanceof ParametersAcceptor) {
```

Comment:
really not sure about this;

the reason i've added it is because in `MutatingScope->enterFunctionLike` calls all phpdoc types are turned to into arguments instead of parameters, which would prevent the templating from occurring, leading the failing test i posted here yesterday.

the idea of this code is to not transform any templates owned by the `callable` or `Closure` phpdoc to arguments, which made my tests and other failing NodeScopeResolver generic tests pass aswell so i think it's correct?

Reply:
had another go at this and it's a can of worms, for example:

```
/**
 * @param Closure<TRetFirst of mixed>(TRetFirst $valone): (Closure<TRetSecond of mixed>(TRetSecond $valtwo): (TRetFirst|TRetSecond)) $closure
 */
function testNestedClosures(Closure $closure, string $str, int $int): void
{
	assertType('Closure<TRetFirst of mixed>(TRetFirst): (Closure<TRetSecond of mixed>(TRetSecond): (TRetFirst|TRetSecond))', $closure);
	$closure1 = $closure($str);
	assertType('Closure<TRetSecond of mixed>(TRetSecond): (string|TRetSecond)', $closure1);
	$result = $closure1($int);
	assertType('string|int', $result);
}
```

can't figure out at all how to handle this in `TypeTraverser::map`. The complexity comes from the fact there are multiple lists of `$ownedTemplates` depending on where in the tree walk we are at, and simply doing